contracts/Treasury.sol

```diff
@@ -95,49 +87,53 @@ contract Treasury is OwnableUpgradeable, ReentrancyGuardUpgradeable, ERC20Upgrad
      * @param _cut uint256     
      */
     function withdraw(address _token, address _creator, uint256 _creatorCounterpart, address _executor, uint256 _executorCounterpart, uint256 _cut) external {
-        require(permissions[STATUS.RESERVESPENDER][msg.sender], notApproved); // check if it's escrow's contract calling
+        require(authorized[msg.sender] == true, "Not approved"); // check if it's escrow's contract calling
       
         // send the cut to the executor based on success rate + his counterpart - the fee
         uint256 executorPayment = _cut + _executorCounterpart - getFee(_executorCounterpart);
-        collectFee(_token, _executorCounterpart);
-        IERC20(_token).transferFrom(address(this), _executor, executorPayment);
+        IERC20(_token).safeTransferFrom(address(this), _executor, executorPayment);
         emit WithdrawERC20(_token, _executor, executorPayment);
 
 
         // send the remainder to offer creator - the fee
         uint256 remainder = _creatorCounterpart - _cut - getFee(_creatorCounterpart);
-        collectFee(_token, _creatorCounterpart); 
-        IERC20(_token).transferFrom(address(this), _creator, remainder);
+        IERC20(_token).safeTransferFrom(address(this), _creator, remainder);
         emit WithdrawERC20(_token, _creator, remainder);
 
     }
+
+    /* ========== GOV ONLY ========== */
     
-     function setPermission(
-        STATUS _status,
-        address _address,
-        bool _permission
-    ) public {
-        permissions[_status][_address] = _permission;
+     function transferGovernor(address _address) public onlyGovernor {
```

Comment:
why are we calling it governor i hate that i keep thinking "guvna" in british accent ;_;

Reply:
we'll change it to admin or keeper or smth